docs/related/async-raml-datatypes.mdx

```diff
@@ -0,0 +1,324 @@
+---
+id: async_raml_datatypes
+title: Async + RAML Data Types in AMF
+---
+
+:::info Technical Document
+This is a technical document, this information is intended for experienced users
+:::
+
+Within the Async specification it is possible to use raml data types to define a payload, taking into account that the schema format is specified accordingly.
+Both specifications have their own distinct referencing and modularization mechanisms, and it is not clear where to draw the line to determine where one specification stops, and the other one starts.
+This document will describe how to use RAML types in Async when using AMF.
+
+
+## Referencing external RAML content using ‘$ref’
+In order to support referencing raml content in external files, AMF allows the use of the `$ref` mechanism present in Async but with clear restrictions.
+This enables referencing data type fragments, data types present in libraries, as well as any raml content present in yaml or json files.
+
+Given that the payload content must comply with RAML, **the ‘$ref’ may only be used at the root of the payload definition as an exception to this rule**.
+Its target content must be valid structurally according to the relevant RAML specification:
+
+export const myStyle = {
+    padding: "15px 5px",
+};
+
+<div className="container">
+<div className='row'>
+<div style={myStyle} className='col col--6'>
+
+```yaml title="‘$ref’ at root of payload (Valid)"
+schemaFormat: application/raml+yaml;version=1.0
+payload:
+  $ref: “...” # must reference raml content
+```
+
+</div>
+<div style={myStyle} className='col col--6'>
+
+```yaml title="‘$ref’ nested within raml content (Invalid)" {6}
+schemaFormat: application/raml+yaml;version=1.0
+payload:
+ type: object
+ properties:
+   user:
+     $ref: “...” # '$ref' is not valid in RAML
+```
+
+</div>
+</div>
+</div>
+
+
+The following section shows all supported use cases when using this referencing mechanism, adding additional comments if necessary.
+
+
+
+### Reference to data type fragments
+The data type fragment must be self-contained, meaning it cannot depend on any definition made in the root api. The following example is **valid** for this use case.
+
+<div className="container">
+<div className='row'>
+<div style={myStyle} className='col col--6'>
+
+```yaml title="Async api"
+schemaFormat: application/raml+yaml;version=1.0
+payload:
+ $ref: external-data-type.raml
+```
+
+</div>
+<div style={myStyle} className='col col--6'>
+
+```yaml title="RAML data type fragment (external-data-type.raml)"
+#%RAML 1.0 DataType
+
+type: object
+properties:
+  a: string | object
+```
+
+</div>
+</div>
+</div>
+
+The following referenced data type is **invalid** as it is not self-contained, meaning the data type fragment depends on type definitions which will not be resolved in the root api.
+Due to known limitations this api will not return validations, but the reference to the type Other will not be made.
```

Comment:
Would it help the end-user to know what these known limitations are? Or would this distract from the scope of this document?

Also, the sentence is a little confusing. Usually "but" implies a contrary condition. Is "the reference to the type Other will not be made" actually contrary to the first part of the sentence? Or is it complementary? If it's complimentary it would be clearer to replace "but" with "and".

Reply:
It's contrary, but I've rewritten this part to make it more clear.